sources/api/shibaken/src/main.rs

```diff
@@ -0,0 +1,278 @@
+/*!
+# Introduction
+
+shibaken is called by sundog as a setting generator.
+
+shibaken will fetch and populate the host container's user-data with authorized ssh keys from the
+AWS instance metadata service (IMDS).
+
+(The name "shibaken" comes from the fact that Shiba are small, but agile, hunting dogs.)
+*/
+
+#![deny(rust_2018_idioms)]
+
+#[macro_use]
+extern crate log;
+
+use reqwest::blocking::Client;
+use serde::Serialize;
+use simplelog::{Config as LogConfig, LevelFilter, TermLogger, TerminalMode};
+use snafu::{ensure, ResultExt};
+use std::str::FromStr;
+use std::{env, process};
+
+// Instance Meta Data Service.
+//
+// Currently only able to get fetch session tokens from `latest`
+// FIXME Pin to a date version that supports IMDSv2 once such a date version is available.
+const IMDS_PUBLIC_KEY_BASE_URL: &str = "http://169.254.169.254/latest/meta-data/public-keys/";
```

Comment:
nit: can omit the trailing slash here and append it in context later so that the format string is more readable